apps/dotcdn/src/app/dotcdn.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { CoreWebService, ResponseView, SiteService } from '@dotcms/dotcms-js';
+import { pluck, mergeMap, map, tap, take } from 'rxjs/operators';
+import * as moment from 'moment';
+import { DotCDNStats, PurgeReturnData, PurgeUrlOptions } from './app.models';
+@Injectable({
+    providedIn: 'root'
+})
+export class DotCDNService {
+    constructor(private coreWebService: CoreWebService, private siteService: SiteService) {}
+
+    /**
+     * Request stats via Core Web Service
+     *
+     * @param {string} period
+     * @return {*}  {Observable<DotCDNStats>}
+     * @memberof DotCDNService
+     */
+    requestStats(period: string): Observable<DotCDNStats> {
+        return this.siteService.getCurrentSite().pipe(
+            take(1),
+            pluck('identifier'),
+            mergeMap((hostId: string) => {
+                const dateTo = moment().format('YYYY-MM-DD');
+                const dateFrom = moment().subtract(period, 'd').format('YYYY-MM-DD');
+                return this.coreWebService.requestView<DotCDNStats>({
+                    url: `/api/v1/dotcdn/stats?hostId=${hostId}&dateFrom=${dateFrom}&dateTo=${dateTo}`
+                });
+            }),
+            pluck('entity')
+        );
+    }
+
+    /**
+     *  Makes a request to purge the cache
+     *
+     * @param {string[]} [urls=[]]
+     * @param {boolean} [invalidateAll=false]
+     * @return {Observable<ResponseView<PurgeReturnData>>}
+     * @memberof DotCDNService
+     */
+    purgeCache(urls?: string[]): Observable<PurgeReturnData> {
+        return this.siteService.getCurrentSite().pipe(
+            pluck('identifier'),
+            mergeMap((hostId: string) => {
+                return this.purgeUrlRequest({ hostId, invalidateAll: false, urls });
+            }),
+            pluck('bodyJsonObject')
```

Comment:
is not `entity` here?

Reply:
No, `purgeUrlRequest` returns an `Observable<ResponseView<PurgeReturnData>>` which has `bodyJsonObject` and other props.